test/test-lib/fake-device.ts

```diff
@@ -54,6 +54,12 @@ export async function patchState(
 	patchBody: StatePatchV3Body,
 	stateVersion: 'v3',
 ): Promise<void>;
+export async function patchState(
+	user: UserObjectParam,
+	deviceUuid: string,
+	patchBody: StatePatchV2Body | StatePatchV3Body,
+	stateVersion: 'v2' | 'v3',
+): Promise<void>;
```

Comment:
This is essentially a duplicate of the final signature and shouldn't be necessary? @thgreasi ?

Reply:
@Page- The final implementation signature isn't considered to be one of the overloads that we can use (iff there is any overload defined). So duplicating it here allows us to pass a variable of type `'v2' | 'v3'` like we have in the test cases.